src/Parser.php

```diff
@@ -408,7 +408,7 @@ private function filterBodyNodes(Node $node, bool $nested = false): ?Node
         // Node::class !== \get_class($node) should be removed in Twig 4.0
         $nested = $nested || (Node::class !== \get_class($node) && !$node instanceof Nodes);
         foreach ($node as $k => $n) {
-            if (null !== $n && null === $this->filterBodyNodes($n, $nested)) {
```

Comment:
$n is never null ; when doing foreach on node, it's `\IteratorAggregate<int|string, Node>`